app/upgrades.go

```diff
@@ -1,3 +1,124 @@
+		UpgradeName_Testnet,
+		upgradeHandler(app, UpgradeName_Testnet),
+	)
+
+	upgradeInfo, err := app.upgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	doUpgrade := upgradeInfo.Name == UpgradeName_Mainnet ||
+		upgradeInfo.Name == UpgradeName_Testnet
+
+	if doUpgrade && !app.upgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added: []string{
+				crisistypes.ModuleName,
+				consensustypes.ModuleName,
+			},
+		}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
+	}
+}
+
+// upgradeHandler returns an UpgradeHandler for the given upgrade parameters.
+func upgradeHandler(
+	app App,
+	name string,
+) upgradetypes.UpgradeHandler {
+	return func(
+		ctx sdk.Context,
+		plan upgradetypes.Plan,
+		fromVM module.VersionMap,
+	) (module.VersionMap, error) {
+		app.Logger().Info(fmt.Sprintf("running %s upgrade handler", name))
+
+		baseAppLegacySS := app.paramsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+
+		// Set param key table for params module migration
+		for _, subspace := range app.paramsKeeper.GetSubspaces() {
+			subspace := subspace
+			var keyTable paramstypes.KeyTable
+			switch subspace.Name() {
+			// sdk
+			case authtypes.ModuleName:
+				keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+			case banktypes.ModuleName:
+				keyTable = banktypes.ParamKeyTable() //nolint:staticcheck,nolintlint
+			case stakingtypes.ModuleName:
+				keyTable = stakingtypes.ParamKeyTable()
+			case minttypes.ModuleName:
+				keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+			case distrtypes.ModuleName:
+				keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck,nolintlint
+			case slashingtypes.ModuleName:
+				keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+			case govtypes.ModuleName:
+				keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+			case crisistypes.ModuleName:
+				keyTable = crisistypes.ParamKeyTable() //nolint:staticcheck
+
+			// ibc
+			case ibctransfertypes.ModuleName:
+				keyTable = ibctransfertypes.ParamKeyTable() //nolint:staticcheck
+
+			default:
+				continue
+			}
+			if !subspace.HasKeyTable() {
+				subspace.WithKeyTable(keyTable)
+			}
+		}
+
+		// optional migration: prune expired tendermint consensus states to save storage space
+		// see https://github.com/cosmos/ibc-go/blob/v7.2.0/docs/migrations/v6-to-v7.md#chains
+		if _, err := ibctmmigrations.PruneExpiredConsensusStates(ctx, app.appCodec, app.ibcKeeper.ClientKeeper); err != nil {
+			return nil, err
+		}
+
+		// migrate tendermint consensus parameters from x/params module to a
+		// dedicated x/consensus module.
+		baseapp.MigrateParams(ctx, baseAppLegacySS, &app.consensusParamsKeeper)
+
+		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
```

Comment:
@DracoLi will this then trigger the cdp migrations to run (as it's registered here https://github.com/Kava-Labs/kava/blob/master/x/cdp/module.go#L126) or do those need to be manually kicked off by calling https://github.com/Kava-Labs/kava/blob/master/x/cdp/migrations/v2/store.go#L11 `MigrateStore` here?

Reply:
this will kick it off here

Reply:
For context, here is the migration log when performing the upgrade
```
7:42PM INF applying upgrade "v0.26.0" at height: 35 module=server
7:42PM INF running v0.26.0 upgrade handler module=server
7:42PM INF pruned expired tendermint consensus states module=x/ibc/client total=0
7:42PM INF migrating module bank from version 3 to version 4 module=server
7:42PM INF migrating module cdp from version 1 to version 2 module=server
7:42PM INF adding a new module: consensus module=server
7:42PM INF migrating module crisis from version 1 to version 2 module=server
7:42PM INF migrating module distribution from version 2 to version 3 module=server
7:42PM INF migrating module gov from version 3 to version 4 module=server
7:42PM INF migrating module ibc from version 2 to version 3 module=server
7:42PM INF migrating module ibc from version 3 to version 4 module=server
7:42PM INF migrating module mint from version 1 to version 2 module=server
7:42PM INF migrating module slashing from version 2 to version 3 module=server
7:42PM INF migrating module staking from version 3 to version 4 module=server
7:42PM INF migrating module transfer from version 2 to version 3 module=server
7:42PM INF migrating module auth from version 3 to version 4 module=server
```